webpack.config.js

```diff
@@ -35,124 +35,60 @@ const sharedConfig = {
-	if ( env.plugin && env.plugin !== 'performance-lab' ) {
-		return defaultBuildConfig;
-	}
-
-	const pluginDir = path.resolve( __dirname, 'plugins/performance-lab' );
-
-	return {
-		...sharedConfig,
-		name: 'performance-lab',
-		plugins: [
-			new CopyWebpackPlugin( {
-				patterns: [
-					{
-						from: `${ pluginDir }/includes/admin/plugin-activate-ajax.js`,
-						to: `${ pluginDir }/includes/admin/plugin-activate-ajax.min.js`,
-					},
-				],
-			} ),
-			new WebpackBar( {
-				name: 'Building Performance Lab Assets',
-				color: '#2196f3',
-			} ),
-		],
-	};
-};
-
-/**
- * Webpack Config: Embed Optimizer
- *
- * @param {*} env Webpack environment
- * @return {Object} Webpack configuration
- */
-const embedOptimizer = ( env ) => {
-	if ( env.plugin && env.plugin !== 'embed-optimizer' ) {
-		return defaultBuildConfig;
-	}
-
-	const pluginDir = path.resolve( __dirname, 'plugins/embed-optimizer' );
-
-	return {
-		...sharedConfig,
-		name: 'embed-optimizer',
-		plugins: [
-			new CopyWebpackPlugin( {
-				patterns: [
-					{
-						from: `${ pluginDir }/detect.js`,
-						to: `${ pluginDir }/detect.min.js`,
-					},
-					{
-						from: `${ pluginDir }/lazy-load.js`,
-						to: `${ pluginDir }/lazy-load.min.js`,
-					},
-				],
-			} ),
-			new WebpackBar( {
-				name: 'Building Embed Optimizer Assets',
-				color: '#2196f3',
-			} ),
-		],
-	};
-};
+const minifyPluginAssets = ( env ) => {
+	if ( env.plugin && ! standalonePlugins.includes( env.plugin ) ) {
+		// eslint-disable-next-line no-console
+		console.error( `Plugin "${ env.plugin }" not found. Aborting.` );
 
-/**
- * Webpack Config: Image Prioritizer
- *
- * @param {*} env Webpack environment
- * @return {Object} Webpack configuration
- */
-const imagePrioritizer = ( env ) => {
-	if ( env.plugin && env.plugin !== 'image-prioritizer' ) {
 		return defaultBuildConfig;
 	}
 
-	const pluginDir = path.resolve( __dirname, 'plugins/image-prioritizer' );
+	const sourcePath = env.plugin
+		? path.resolve( __dirname, 'plugins', env.plugin )
+		: path.resolve( __dirname, 'plugins' );
 
 	return {
 		...sharedConfig,
-		name: 'image-prioritizer',
+		name: 'minify-plugin-assets',
 		plugins: [
 			new CopyWebpackPlugin( {
 				patterns: [
 					{
-						from: `${ pluginDir }/lazy-load-video.js`,
-						to: `${ pluginDir }/lazy-load-video.min.js`,
-					},
-					{
-						from: `${ pluginDir }/lazy-load-bg-image.js`,
-						to: `${ pluginDir }/lazy-load-bg-image.min.js`,
+						from: `${ sourcePath }/**/*.js`,
+						noErrorOnMissing: true,
+						to: ( { absoluteFilename } ) =>
+							absoluteFilename.replace( /\.js$/, '.min.js' ),
+						// Exclude already-minified files and those in the build directory
+						globOptions: {
+							ignore: [ '**/build/**', '**/*.min.js' ],
+						},
 					},
 					{
-						from: `${ pluginDir }/lazy-load-bg-image.css`,
-						to: `${ pluginDir }/lazy-load-bg-image.min.css`,
+						from: `${ sourcePath }/**/*.css`,
+						noErrorOnMissing: true,
```

Comment:
The `noErrorOnMissing: true` option prevents build errors for plugins without JavaScript or CSS files during plugin-specific minification.